### PR TITLE
Sync: Auth

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206519747938382/f

### Description
Add system wide authentication to Sync related actions
Remove Beta info panel from Settings screen
Remove QR code from sync enabled screen, replace it with Sync with another device
Update Delete Server Data dialog

### Steps to test this PR

_Create account_
- [ ] Open Sync Settings screen
- [ ] Tap on create account
- [ ] Verify device auth appears

_Recover account_
- [ ] Open Sync Settings screen
- [ ] Tap on recover account
- [ ] Verify device auth appears

_Sync with another device_
- [ ] Open Sync Settings screen
- [ ] Tap on sync with another device
- [ ] Verify device auth appears

_Sync enabled - Sync with another device_
- [ ] Enable Sync
- [ ] Open Sync Settings screen
- [ ] Tap on sync with another device
- [ ] Verify device auth appears

_Sync enabled -Recovery PDF_
- [ ] Enable Sync
- [ ] Open Sync Settings screen
- [ ] Tap on Recover PDF
- [ ] Verify device auth appears

_Grace period_
- [ ] Open Sync Settings screen
- [ ] Tap on create account
- [ ] Verify device auth appears
- [ ] Authenticate
- [ ] In Sync Settings screen, tap on recover pdf
- [ ] Verify authentication is not required
- [ ] Wait 15s or more
- [ ] In Sync Settings screen, tap on recover pdf
- [ ] Verify authentication is required again

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206534576482448

### CODE REVIEW

**Comment by malmstein** [comment]
* `develop`
  * **#4141** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4141?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/SupportedDeviceAuthChecker.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import android.app.KeyguardManager
+import android.content.Context
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricManager.Authenticators
+import androidx.biometric.BiometricManager.BIOMETRIC_SUCCESS
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface SupportedDeviceAuthChecker {
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/SupportedDeviceAuthChecker.kt:28]: This is used from `hasValidDeviceAuthentication`,  which seems unused. do we tell the user that needs to have auth enabled?

**Review by cmonfortep — COMMENTED** [review]
Took a pass on the code. Can you review all the autofill copied code? there are many references to autofill (also classname reference to autofill), unused methods, and I wonder if we need the same interfaces in our usecase.

Testing I noticed we don't tell the user that needs to configure an authentication method. I think autofill has this part covered. Will comment on Asana.

Left some comments about screens we don't need anymore (please check the manifest).

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/SupportedDeviceAuthChecker.kt (hunk 1 above)
Comment (malmstein) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/SupportedDeviceAuthChecker.kt:28]: This is no longer a problem. If the user doesn’t have auth enabled we will just go ahead for now.

**Comment by malmstein** [comment]
Thanks for the review @cmonfortep . I’ve addressed the comments and push them, can you take a look?

**Review by cmonfortep — COMMENTED** [review]
I haven't found anything off on the code, but when testing found one issue with device names. Will post in the Ship review task.

**Review by cmonfortep — APPROVED** [review]
LGTM thanks!

Please remember to force dark mode on the new screen 🙏

**Pull request merged**